app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,139 @@
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $categories = Category::all();
+        return view('upload-product',compact('categories'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $request->validate([
+            'name' => ['required'],
+            'description' => ['required'],
+            'price' => ['required'],
+            'category' => ['required'],
+            'path' => ['required','mimes:jpg,jpeg,gif,png,pdf']
+        ]);
+
+        $product_model = new Product;
+
+        if($request->file('path')){
+            $file_name = time().'_'.$request->file('path')->getClientOriginalName();
+            $file_path = $request->file('path')->storeAs('post-uploads', $file_name, 'public');
+
+            $product_model->name = $request->name;
+            $product_model->description = $request->description;
+            $product_model->price = $request->price;
+            $product_model->category_id = $request->category;
+            $product_model->path = '/storage/' . $file_path;
+            $product_model->units = $request->units;
+
+            $product_model->save();
+            return redirect()->route('products.index')->with('success','Product created successfully.');
+
+        }
+
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        //
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit(Product $product)
+    {
+        return view('edit-image',compact('product'));
+
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request,Product $product)
+    {
+        $request->validate([
+            'name' => ['required'],
+            'description' => ['required'],
+            'price' => ['required']
+        ]);
+
+
+            // $file_name = time().'_'.$request->file('path')->getClientOriginalName();
+            // $file_path = $request->file('path')->storeAs('post-uploads', $file_name, 'public');
+
+
+            $product->update(['name'=> $request->name]);
+            $product->update(['description'=> $request->description]);
+            $product->update(['price'=> $request->price]);
+            $product->update(['units'=> $request->units]);
+
+            return redirect()->route('products.index');
```

Comment:
Please add the with() method here with success message: Product  updated successfully